### PR TITLE
Fix an error when Cybran air factories are destroyed while they are building a unit

### LINKED ISSUE #5796: Bug in effectutilities.lua
Bug 3778. Need to add a check for nil or on unit death..
```
warning: Error running lua script: ...a\faforever\gamedata\lua.nx2\lua\effectutilities.lua(345): attempt to loop over field `BuildEffectBones' (a nil value)
         stack traceback:
         	...a\faforever\gamedata\lua.nx2\lua\effectutilities.lua(345): in function <...a\faforever\gamedata\lua.nx2\lua\effectutilities.lua:341>
         	...ata\lua.nx2\lua\sim\units\cybran\cairfactoryunit.lua(62): in function <...ata\lua.nx2\lua\sim\units\cybran\cairfactoryunit.lua:56>
         	```

### PR DESCRIPTION
**Pull request by Basilisk3**
Closes #5796

### CODE REVIEW

File: lua/EffectUtilities.lua [hunk 1]
```diff
@@ -340,11 +340,14 @@ local UnitBuildEffects = {
 ---@param buildEffectsBag TrashBag
 function CreateCybranFactoryBuildEffects(builder, unitBeingBuilt, buildBones, buildEffectsBag)
     CreateCybranBuildBeamsOpti(builder, nil, unitBeingBuilt, buildEffectsBag, false)
+    WaitTicks(10)
 
     local builderArmy = builder.Army
-    for _, bone in buildBones.BuildEffectBones do
-        for _, effect in BuildEffects do
-            TrashBagAdd(buildEffectsBag, CreateAttachedEmitter(builder, bone, builderArmy, effect))
+    while not unitBeingBuilt:BeenDestroyed() do
```
Comment (Garanas) [inline lua/EffectUtilities.lua:346]: I'm fairly certain that this should be an `if` statement and not a while loop.

File: lua/EffectUtilities.lua [hunk 2]
```diff
@@ -340,11 +340,14 @@ local UnitBuildEffects = {
 ---@param buildEffectsBag TrashBag
 function CreateCybranFactoryBuildEffects(builder, unitBeingBuilt, buildBones, buildEffectsBag)
     CreateCybranBuildBeamsOpti(builder, nil, unitBeingBuilt, buildEffectsBag, false)
+    WaitTicks(10)
```
Comment (Garanas) [inline lua/EffectUtilities.lua:343]: Why would we wait here?

File: lua/EffectUtilities.lua (hunk 1 above)
Comment (Basilisk3) [reply lua/EffectUtilities.lua]: 👍

File: lua/EffectUtilities.lua (hunk 2 above)
Comment (Basilisk3) [reply lua/EffectUtilities.lua]: I reduced it to 8, unfortunately some sort of delay appears to be required. At 5 ticks, it still errors out sometimes.

**Comment by Garanas** [comment]
I'm not certain whether this is the right direction. The additional wait is 'fixing' something else, but I feel like we do not fully understand what the wait fixes.

**Comment by Garanas** [comment]
I don't think this is the correct direction to take. The error is not critical, I'll try another approach in a bit when I re-evaluate the Cybran build effects in general.

**Pull request closed without merging**